lib/checkunusedvar.cpp

```diff
@@ -1267,7 +1267,8 @@ void CheckUnusedVar::checkFunctionVariableUsage()
                 continue;
 
             FwdAnalysis fwdAnalysis(mTokenizer->isCPP(), mSettings->library);
-            if (fwdAnalysis.unusedValue(expr, start, scope->bodyEnd)) {
+            const Token* scopeEnd = expr->variable() ? getEndOfVarScope(expr, { expr->variable() }) : scope->bodyEnd;
```

Comment:
It would be better to use [`getVariables`](https://github.com/danmar/cppcheck/blob/main/lib/valueflow.cpp#L5072) to get the end scope of all variables in the expression.

Reply:
You mean instead of `{ expr->variable() }`?

Reply:
Yes